web-ng/src/app/shared/models/observation/observation.model.ts

```diff
@@ -28,4 +28,15 @@ export class Observation {
     readonly lastModified: AuditInfo,
     readonly responses: Map<string, Response>
   ) {}
+
+  copy(responses: Map<string, Response>) {
```

Comment:
I'd expect a method called "copy" to only copy an object to another of the same class, whereas this is actually copying and modifying repsonses. I think the idiomatic way of calling this might be "withResponses()".